packages/lit-html/src/directives/spread.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {noChange, nothing, _Σ as litHtmlPrivate} from '../lit-html.js';
+import {
+  directive,
+  Directive,
+  DirectiveParameters,
+  PartInfo,
+  PartType,
+  Part,
+  ElementPart,
+} from '../directive.js';
+
+const {
+  _AttributePart: AttributePartImpl,
+  _PropertyPart: PropertyPartImpl,
+  _BooleanAttributePart: BooleanAttributePartImpl,
+  _EventPart: EventPartImpl,
+  // _ElementPart: ElementPartImpl,
+} = litHtmlPrivate;
+
+export interface SpreadValues {
+  readonly [name: string]: unknown;
+}
+
+class SpreadDirective extends Directive {
```

Comment:
We should consider whether this should be an `AsyncDirective` so that its `disconnected` callback can undo all the parts if you render a different value to the expression (or whether spread must be static), e.g....

```html
<div ${bool ? spread({foo, bar}) : nothing}></div>
``` 

To be fair, we don't do a good thing for e.g. `classMap` in this situation either (it would also need to be disconnectable), which seemed a bit overkill given the performance hit:
```html
<div class=${bool ? classMap({foo, bar}) : nothing}></div>
```

Reply:
I agree this is the most complete implementation, but given that it's more costly it might be better to keep things simple. I think this pattern _should_ suffice:

```
<div ${spread(bool ? {foo, bar} : undefined)}></div>
```